### PR TITLE
chore: Improve execution time logging

### LINKED ISSUE #11577: TimeoutException: Did not observe any item or terminal signal within 10000ms in 'source(MonoCallable)' (and no fall...
Sentry Issue: [APPSMITH-1BP](https://sentry.io/organizations/appsmith/issues/2782006514/?referrer=github_integration)

```
TimeoutException: Did not observe any item or terminal signal within 10000ms in 'source(MonoCallable)' (and no fallback has been configured)
    at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.handleTimeout(FluxTimeout.java:295)
```

### LINKED ISSUE #13810: Log execution time on mixpanel

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> As a part of improving the performance for the import-export flows we want to log the data on mixpanel so that we can device a strategy for improving the performance. Today as the data is not logged in mixpanel it's difficult to pinpoint the performance degradation. This PR will log the execution time for both import and export flows to mixpanel and also includes the improvement in the console logs to debug on production env.  

Fixes #13810 
Fixes #11577

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Manual testing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java [hunk 1]
```diff
@@ -132,7 +132,6 @@ public Mono<Path> saveApplicationToGitRepo(Path baseRepoSuffix,
         // Checkout to mentioned branch if not already checked-out
         Stopwatch processStopwatch = new Stopwatch("FS application save");
         return gitExecutor.resetToLastCommit(baseRepoSuffix, branchName)
-                .then(gitExecutor.checkoutToBranch(baseRepoSuffix, branchName))
                 .flatMap(isSwitched -> {
```
Comment (AnaghHegde) [inline app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java:135]: Why have we removed this ?

File: app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    CLONE,
+    CREATE_BRANCH,
+    DELETE_BRANCH,
+    CHECKOUT,
+    SYNC_BRANCH,
+    LIST_LOCAL_BRANCH,
+    MERGE,
+    MERGE_CHECK,
+    FETCH
+}
```
Comment (AnaghHegde) [inline app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java:19]: Why can't we use the existing class `AnalyticsEvents.java` ? We can move the class to interface module if it causing the dependency issues.

**Review by AnaghHegde — COMMENTED** [review]
Minor comments. Rest all looks good to me.

File: app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java (hunk 2 above)
Comment (abhvsn) [reply app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java:19]: Actual git level actions may alter compared to the server service class, so thought of creating a new constants file in git module. Also partly this was related to the analytics class dependency but as the change was small enough I though of keeping this in server module for now.

File: app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java (hunk 1 above)
Comment (abhvsn) [reply app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java:135]: We are already doing this in `gitExecutor.resetToLastCommit(baseRepoSuffix, branchName)` so removed the duplication.

File: app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java (hunk 2 above)
Comment (AnaghHegde) [reply app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java:19]: Yes that is true. But i thought that we would have to maintain this information in both the places. So one place is better right?

Comment (abhvsn) [reply app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java:19]: Sounds good let me fix this

Comment (AnaghHegde) [reply app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java:19]: Just a thought, if we use the same name would it cause the duplicates in git events on mix panel ?

Comment (abhvsn) [reply app/server/appsmith-git/src/main/java/com/appsmith/git/constants/GitActions.java]: There are 2 different locations we are logging the events:
1. Server module 
2. Git module
On server module events are logged on mixpanel, but on git side we are logging these on server logs so these will not cause any conflicts

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**